Core/include/Acts/EventData/detail/CorrectedTransformationFreeToBound.hpp

```diff
@@ -0,0 +1,111 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021-2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Common.hpp"
+#include "Acts/EventData/detail/TransformationFreeToBound.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+namespace Acts {
+
+/// @brief Free to bound transformation Correction configuration class
+///
+struct FreeToBoundCorrection {
+  /// Apply correction or not
+  bool apply = false;
+
+  /// UKF tuning parameters
+  ActsScalar alpha = 0.1;
+  ActsScalar beta = 2;
+
+  /// The cutoff of incident angles cosine for correction
+  ActsScalar cosIncidentAngleMinCutoff = 1e-5;
+  ActsScalar cosIncidentAngleMaxCutoff = 0.99500417;
+
+  /// Default constructor
+  FreeToBoundCorrection() = default;
+
+  /// Construct from boolean and UKF parameters (alpha, beta)
+  ///
+  /// @param apply_ Wheter to apply correction
+  /// @param alpha_ The UKF tuning parameter alpha
+  /// @param beta_ The UKF tuning parameter beta
+  FreeToBoundCorrection(bool apply_, ActsScalar alpha_, ActsScalar beta_);
+
+  /// Construct from boolean only
+  ///
+  /// @param apply_ Wheter to apply correction
+  FreeToBoundCorrection(bool apply_);
```

Comment:
@asalzburger @XiaocongAi I'm sorry maybe our previous discussion wasn't entirely clear. This constructor is what I was unhappy with. Can we make this

```cpp
explicit FreeToBoundCorrection(bool apply_);
```

so the autoconversion from `bool` will be rejected by the compiler right away?

Sorry I didn't see earlier that you only changed the other location I commented on.

Reply:
Thank you! Fixed now.